sdk/src/main/java/com/qonversion/android/sdk/QProductCenterManager.kt

```diff
@@ -1,4 +1,72 @@
 package com.qonversion.android.sdk
 
-class QProductCenterManager {
+import android.app.Application
+import android.util.Pair
+import com.android.billingclient.api.Purchase
+import com.android.billingclient.api.SkuDetails
+import com.qonversion.android.s.LifecycleCallback
+import com.qonversion.android.sdk.ad.AdvertisingProvider
+import com.qonversion.android.sdk.billing.Billing
+import com.qonversion.android.sdk.converter.GooglePurchaseConverter
+import com.qonversion.android.sdk.converter.PurchaseConverter
+import com.qonversion.android.sdk.extractor.SkuDetailsTokenExtractor
+import com.qonversion.android.sdk.logger.Logger
+
+class QProductCenterManager internal constructor(
+    private var repository: QonversionRepository,
+    private var logger: Logger
+) {
+    @Volatile
+    var billingClient: Billing? = null
+        @Synchronized private set
+        @Synchronized get
+
+    @Volatile
+    var billing: QonversionBilling? = null
+        @Synchronized private set
+        @Synchronized get
+
+    @Volatile
+    lateinit var converter: PurchaseConverter<Pair<SkuDetails, Purchase>>
+        @Synchronized private set
+        @Synchronized get
+
+    fun launch(
+        context: Application,
+        billingBuilder: QonversionBillingBuilder?,
+        callback: QonversionCallback?
+    ) {
+        converter = GooglePurchaseConverter(SkuDetailsTokenExtractor())
+        val adProvider = AdvertisingProvider()
+        adProvider.init(context, object : AdvertisingProvider.Callback {
+            override fun onSuccess(advertisingId: String) {
+                repository.init(advertisingId, callback)
+            }
+
+            override fun onFailure(t: Throwable) {
+                repository.init(callback)
+            }
+        })
+
+        billing = if (billingBuilder != null) {
+            QonversionBilling(context, billingBuilder, logger, true)
+        } else {
+            null
+        }
+        billing?.setReadyListener { purchase, details ->
+            purchase(Pair.create(details, purchase), null)
+        }
+        billingClient = billing
+
+        val lifecycleCallback = LifecycleCallback(repository)
+        context.registerActivityLifecycleCallbacks(lifecycleCallback)
+    }
+
+    fun purchase(
+        purchaseInfo: android.util.Pair<SkuDetails, Purchase>,
+        callback: QonversionCallback?
+    ) {
+        val purchase = converter.convert(purchaseInfo)
+        repository.purchase(purchase, callback)
+    }
```

Comment:
As far as we don't use converter in other places, maybe we should store it like local variable?